Clarify swap descriptions (#14480)
pingcap/docs: check-before-deployment.md

## Check and disable system swap

> TiDB needs sufficient memory space for operation. If you want to maintain stable performance, it is recommended that you permanently disable the system swap, but it might trigger OOM issues when there is insufficient memory. If you want to avoid such OOM issues, you can just decrease the swap priority, instead of permanently disabling it.

Reviewer: ```suggestion
TiDB needs a sufficient amount of memory for operation. If the memory that TiDB uses gets swapped out and later gets swapped back in this can cause latency spikes. If you want to maintain stable performance, it is recommended that you permanently disable the system swap, but it might trigger OOM issues when there is insufficient memory. If you want to avoid such OOM issues, you can just decrease the swap priority, instead of permanently disabling it.
```

- Enabling and using swap might introduce performance jitter issues. It is recommended that you permanently disable the operating system tier swap for low-latency and stability-critical database services. To permanently disable swap, you can use the following method:

    - During the initialization phase of the operating system, do not partition the swap partition disk separately.
    - If you have already partitioned a separate swap partition disk during the initialization phase of the operating system and enabled swap, run the following command to disable it:

        ```bash
        echo "vm.swappiness = 0">> /etc/sysctl.conf

>         swapoff -a

Reviewer: This is not sufficient. The `swapoff -a` command only disables the swap until the next reboot of the system. One also needs to remove this from `/etc/fstab` of mask the systemd unit for swap.

Author: @bb7133 PTAL

Reviewer: https://rageagainstshell.com/2019/12/disable-swap-in-systemd/ has a good description for systemd based systems

Reviewer: For now I would suggest to keep it closer to the original statement:

```bash
echo "vm.swappiness = 0">> /etc/sysctl.conf
sysctl -p
swapoff -a && swapon -a
```

This sets `swappiness` to 0 and then disables/enables swap to for things that have already been swapped out to be swapped back in.

If we leave swap disabled, but don't configure this in a way that survives a reboot this is like leaving a booby trap.

>         echo "vm.swappiness = 0">> /etc/sysctl.conf
>         swapoff -a

Author: ```suggestion
        echo "vm.swappiness = 0">> /etc/sysctl.conf
        sysctl -p
        swapoff -a && swapon -a
```

>         sysctl -p

Author: ```suggestion
```
